module/Company/src/Company/Service/Company.php

```diff
@@ -160,7 +173,10 @@ public function getCategoryList($visible)
                     $translator->translate('You are not allowed to access the admin interface')
                 );
             }
-            return $this->getCategoryMapper()->findAll();
+            $results = $this->getCategoryMapper()->findAll();
```

Comment:
Why not just add a `distinct` clause to the query here? Wouldn't that solve all of this in 1 line of code?